app/services/users.py

```diff
@@ -168,15 +167,25 @@ async def delete_user_read_states(current_user: User):
 
 
 async def get_users_info(data: dict):
-    user_ids: List[str] = data.get("user_ids", [])
-    user_obj_ids = []
-    for user_id in user_ids:
-        try:
-            user_obj_ids.append(ObjectId(user_id))
-        except InvalidId:
-            pass
-
-    return await get_items(filters={"_id": {"$in": user_obj_ids}}, result_obj=User)
+    users = []
+
+    user_ids: List[str] = [ObjectId(i) for i in data.get("user_ids", []) if ObjectId.is_valid(i)]
+
+    if len(user_ids) > 0:
+        id_users = await get_items(filters={"_id": {"$in": user_ids}}, result_obj=User, limit=None)
+        users.extend(id_users)
+
+    wallet_addresses: List[str] = [
+        checksum_address(a) for a in data.get("wallet_addresses", []) if is_account_address(a)
+    ]
+
+    if len(wallet_addresses) > 0:
+        address_users = await get_items(
+            filters={"wallet_address": {"$in": wallet_addresses}}, result_obj=User, limit=None
+        )
+        users.extend(address_users)
+
+    return users
```

Comment:
Not super important, but this might return the same User object twice in the JSON.

Say we have a User(id=1, wallet_address=0x2). If we call this endpoint with `user_ids: ['1', '0x2']` it'd append the same User twice to the `users` object

Reply:
Hmm, you’re right! 🤔

Reply:
I don’t immediately know how to fix that so way so I’ll just leave a note for now 😅🙈